ggml-rpc.cpp

```diff
@@ -273,6 +282,22 @@ static bool send_rpc_cmd(const std::shared_ptr<socket_t> & sock, enum rpc_cmd cm
 
 // RPC client-side implementation
 
+static void free_rpc_backend(rpc_backend_ptr rpc_back) {
+    ggml_backend_t backend = rpc_back->backend;
+    ggml_backend_rpc_context * rpc_ctx = (ggml_backend_rpc_context *)backend->context;
+    std::string endpoint = rpc_back->endpoint;
+    ggml_backend_rpc_buffer_type_context * buft_ctx = (ggml_backend_rpc_buffer_type_context *)rpc_ctx->buft->context;
+    GGML_PRINT_DEBUG("[%s] closing connection to %s\n", __func__, endpoint.c_str());
+    delete buft_ctx;
+    delete rpc_ctx->buft;
+    delete rpc_ctx;
+    delete backend;
+    instances.erase(endpoint);
```

Comment:
should we lock the `instances` here? will we access it from different thread simultaneously?